### PR TITLE
Feast session 01-1

### PR DESCRIPTION
**Pull request by nicolas33**
Learn what was changed in stateController and pass changed collection to the other side.

### CODE REVIEW

File: poc-07.py [hunk 1]
```diff
@@ -84,7 +87,16 @@ def search(self):
...
             else:
-                pass # TODO: compare: did message change?
+                for stateMessage in stateMessages:
+                    if message.uid == stateMessage.uid:
+                        if not message.identical(stateMessage):
+                            changedMessages.append(message)
+
+        for stateMessage in stateMessages:
+            if stateMessage not in messages:
+                # TODO: mark message as destroyed from real repository.
```
Comment (iamprakashom) [inline poc-07.py:99]: what is real repository here? left or right?

File: poc-07.py [hunk 2]
```diff
@@ -68,7 +89,16 @@ def search(self):
                 # Missing in the other side.
```
Comment (Deepanshu2017) [inline poc-07.py]: This loop will run in O(N^2) (in worse case) can we improve it?

File: poc-07.py [hunk 3]
```diff
@@ -68,7 +89,16 @@ def search(self):
...
+                    if message.uid == stateMessage.uid:
+                        if not message.identical(stateMessage):
+                            changedMessages.append(message)
+
+        for stateMessage in stateMessages:
+            if stateMessage not in messages:
+                # TODO: mark message as destroyed from real repository.
+                changedMessages.append(message)
+
         return changedMessages
```
Comment (Deepanshu2017) [inline poc-07.py:103]: Above loop will run in O(N^2) in worse case. Can we improve it?

File: poc-07.py (hunk 1 above)
Comment (Deepanshu2017) [inline poc-07.py]: Depends on the object calling it

File: poc-07.py [hunk 4]
```diff
@@ -96,7 +106,8 @@ def run(self):
     m1l = Message(1, "1 body") # Same as m1r
 
     m2r = Message(2, "2 body")
-    m2l = Message(2, "2 body", ['read']) # Same as m2r but read.
+    m2l = Message(2, "2 body")
+    m2l.markRead()              # Same as m2r but read.
```
Comment (iamprakashom) [inline poc-07.py]: good!

File: poc-07.py (hunk 1 above)
Comment (nicolas33) [inline poc-07.py:99]: Each stateController is attached on one side (and for both). Hence, each stateController acts as the owner and the stranger of one driver.

File: poc-07.py (hunk 3 above)
Comment (nicolas33) [inline poc-07.py:103]: That's all I could come with for now. I wonder we could improve this by improving the Messages (collection) class.

**Pull request closed without merging**